### PR TITLE
Anomaly detection module

### PR DESCRIPTION
**Pull request by VoidDruid**
_work in progress, right now this PR only for review and CI pipeline cheks_

Linked to issue(s): first step to closing [anomaly detection issue](https://github.com/astrolabsoftware/fink-science/issues/264)

## If this is a new release, did you issue the corresponding schema in fink-client?

PR in broker - https://github.com/astrolabsoftware/fink-broker/pull/554

## What changes were proposed in this pull request?

Add `anomaly_score` column for anomaly detection pipeline.

## How was this patch tested?

Doctests, in jupyter with fink test data, and manual testing.

### CODE REVIEW

**Review by JulienPeloton — CHANGES_REQUESTED** [review]
Thanks @VoidDruid for the new module! Some comments on the models:
- models should be put in the folder `fink_science/data/models` with all the others (and the path should be updated accordingly)
- the two models together are about 40MB. This repository, with models, is packaged and distributed on pypi which has a hard limit of 100MB for each repository. Is there any chance you can make your models slimmer?

File: fink_science/anomaly_detection/processor.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        self.forest_g = forest_g
+
+    def anomaly_score(self, data_g, data_r):
+        scores_g = self.forest_g.score_samples(data_g)
+        scores_r = self.forest_r.score_samples(data_r)
+        return (scores_g + scores_r) / 2
+
+
+path = os.path.dirname(__file__)
+with open(f"{path}/anomaly_detection/forest_r.pickle", 'rb') as forest_file:
```
Comment (JulienPeloton) [inline fink_science/anomaly_detection/processor.py:51]: Change the path to `data/models` using:

```python
from fink_science import __file__

...
path = os.path.dirname(os.path.abspath(__file__))
model_path = curdir + '/data/models/forest_r.pickle'
with open(model_path, 'rb') as forest_file:
  ...
...
```

File: fink_science/anomaly_detection/processor.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    forest_r = pickle.load(forest_file)
+with open(f"{path}/anomaly_detection/forest_g.pickle", 'rb') as forest_file:
```
Comment (JulienPeloton) [inline fink_science/anomaly_detection/processor.py:53]: Change the path to data/models using:

```python
from fink_science import __file__

...
path = os.path.dirname(os.path.abspath(__file__))
model_path = curdir + '/data/models/forest_r.pickle'
with open(model_path, 'rb') as forest_file:
  ...
```

Comment (VoidDruid) [reply fink_science/anomaly_detection/processor.py]: Done

**Comment by VoidDruid** [comment]
>      models should be put in the folder `fink_science/data/models` with all the others (and the path should be updated accordingly)

Done

>      the two models together are about 40MB. This repository, with models, is packaged and distributed on pypi which has a hard limit of 100MB for each repository. Is there any chance you can make your models slimmer?

Put pickle files into a zip archive, which will unzip on first module import, zip archive is ~13mb.

**Review by JulienPeloton — APPROVED** [review]
Thanks @VoidDruid ! All good to me, I will merge this PR, and move to the PR in fink-broker to add the module.

**Pull request merged**